drivers/gles3/storage/material_storage.cpp

```diff
@@ -2597,7 +2597,7 @@ void CanvasShaderData::set_code(const String &p_code) {
 
 	actions.uniforms = &uniforms;
 	Error err = MaterialStorage::get_singleton()->shaders.compiler_canvas.compile(RS::SHADER_CANVAS_ITEM, code, &actions, path, gen_code);
-	ERR_FAIL_COND_MSG(err != OK, "Shader compilation failed.");
+	ERR_FAIL_COND_MSG(err != OK, "CanvasItem shader compilation failed (most likely due to a syntax error). Shader code has been printed nearby for troubleshooting.");
```

Comment:
```suggestion
	ERR_FAIL_COND_MSG(err != OK, "CanvasItem shader compilation failed (most likely due to a syntax error). Shader code has been printed above for troubleshooting.");
```

I think "above" makes more sense than "nearby", so I would suggest you update all the messages.